src/lib/page_scripts/trade_offers.ts

```diff
@@ -1,15 +1,96 @@
 
 init('src/lib/page_scripts/trade_offers.js', main);
 
 function main() {}
 
+/**
+ * Gets the trade offers from the local storage or fetches them from the API.
+ * Local storage serves as a cache here.
+ * @param isSentPage if the current page is the sent trade offers page
+ * @returns the trade offers
+ */
+async function fetchTradeOffers(isSentPage: boolean) {
+    const g_steamTrades = JSON.parse(localStorage.getItem('g_steamTrades') || '{}') as FetchSteamTradesResponse;
+    let refetchRequired = true;
+    if (g_steamTrades.sent || g_steamTrades.received) {
+        const latestTradeId = Number.parseInt(g_steamTrades[isSentPage ? 'sent' : 'received']?.[0].offer_id);
+        const latestTradeElement = Number.parseInt(document.querySelector('.tradeoffer')?.id.split('_')[1] ?? '0');
+
+        refetchRequired = Number.isNaN(latestTradeId) || latestTradeId !== latestTradeElement;
+    }
+
+    if (!refetchRequired) {
+        return g_steamTrades;
+    }
+
+    const steamTrades = await ClientSend(FetchSteamTrades, {});
+
+    localStorage.setItem('g_steamTrades', JSON.stringify(steamTrades));
+    return steamTrades;
+}
+
+/**
+ * Fetches the api data for trade offers and stores relevant data in the DOM to be used by Lit components.
+ */
+async function getAndStoreTradeOffers() {
+    const isSentPage = location.pathname.includes('sent');
+
+    const steamTrades = await fetchTradeOffers(isSentPage);
+
+    const tradeOffers = document.querySelectorAll('.tradeoffer');
+
+    for (const tradeOffer of tradeOffers) {
+        const tradeId = tradeOffer.id.split('_')[1];
+        const tradeItems = tradeOffer.querySelectorAll('.trade_item');
+        const trade = isSentPage
+            ? steamTrades.sent.find((t) => t.offer_id === tradeId)
+            : steamTrades.received.find((t) => t.offer_id === tradeId);
+
+        for (const tradeItem of tradeItems) {
+            const economyItemParts = tradeItem.getAttribute('data-economy-item')?.split('/');
+            const classId = economyItemParts?.[2];
+            const instanceId = economyItemParts?.[3];
+
+            if (!instanceId) {
+                continue;
+            }
+
+            const description = steamTrades.descriptions.find(
+                (d) => d.classid === classId && d.instanceid === instanceId
+            );
+            if (description) {
+                tradeItem.setAttribute('data-description', JSON.stringify(description));
+            }
+
+            let isOwnItem = true;
+            let apiItem = trade?.given_asset_ids?.find((a) => a.classid === classId && a.instanceid === instanceId);
+            if (!apiItem) {
+                isOwnItem = false;
+                apiItem = trade?.received_asset_ids?.find((a) => a.classid === classId && a.instanceid === instanceId);
+            }
+            const ownerId = isOwnItem
+                ? JSON.parse(document.getElementById('application_config')?.dataset?.userinfo || '{}').steamid
```

Comment:
Let's move getting the logged in users' Steam ID from the page into a utility function that can be used elsewhere.

We generally parse `g_steamID` from the page instead of this application config, which is IMO less _likely_ to change than the application config (which is a more recent change on steam). I don't have strong opinions on this however.

Reply:
I don't think we have access to the JS-environment of the main page in page scripts. At least it didn't work in my tests:
```
trade_offers.ts:82 Uncaught (in promise) ReferenceError: g_steamID is not defined
```
So for now I moved `application-config`-parsing into its own utility function.